sdk/nodejs/tests/runtime/closure.spec.ts

```diff
@@ -4699,6 +4770,217 @@ return function () { console.log(o1); console.log(o2.b.d); console.log(o3.b.d);
         });
     }
 
+    {
+        // Used just to validate that if we capture a Config object we see these values serialized over.
+        // Specifically, the module that Config uses needs to be captured by value and not be
+        // 'require-reference'.
+        runtime.setConfig("test:TestingKey1", "TestingValue1");
+        const testConfig = new config.Config("test");
+
+        cases.push({
+            title: "Capture config created on the outside",
+            func: function () { const v = testConfig.get("TestingKey1"); console.log(v); },
+            expectText: `exports.handler = __f0;
+
+var __testConfig_proto = {};
+__f1.prototype = __testConfig_proto;
+Object.defineProperty(__testConfig_proto, "constructor", { configurable: true, writable: true, value: __f1 });
+var __config = {["test:TestingKey1"]: "TestingValue1", ["test:TestingKey2"]: "TestingValue2"(...)
+var __runtime_1 = {getConfig: __getConfig};
+Object.defineProperty(__testConfig_proto, "get", { configurable: true, writable: true, value: __f2 });
+__f5.isInstance = __f6;
+(...)
+Object.setPrototypeOf(__f4, __f5);
+Object.defineProperty(__testConfig_proto, "getBoolean", { configurable: true, writable: true, value: __f3 });
+Object.defineProperty(__testConfig_proto, "getNumber", { configurable: true, writable: true, value: __f7 });
+Object.defineProperty(__testConfig_proto, "getObject", { configurable: true, writable: true, value: __f8 });
+Object.setPrototypeOf(__f10, __f5);
+Object.defineProperty(__testConfig_proto, "require", { configurable: true, writable: true, value: __f9 });
+Object.defineProperty(__testConfig_proto, "requireBoolean", { configurable: true, writable: true, value: __f11 });
+Object.defineProperty(__testConfig_proto, "requireNumber", { configurable: true, writable: true, value: __f12 });
+Object.defineProperty(__testConfig_proto, "requireObject", { configurable: true, writable: true, value: __f13 });
+Object.defineProperty(__testConfig_proto, "fullKey", { configurable: true, writable: true, value: __f14 });
+var __testConfig = Object.create(__testConfig_proto);
+__testConfig.name = "test";
+(...)
+function __cleanKey() {
+  return (function() {
+    with({ cleanKey: __cleanKey }) {
+
+return function /*cleanKey*/(key) {
+    const idx = key.indexOf(":");
+    if (idx > 0 && key.startsWith("config:", idx + 1)) {
+        return key.substring(0, idx) + ":" + key.substring(idx + 1 + "config:".length);
+    }
+    return key;
+};
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
+}
+
+function __ensureConfig() {
+  return (function() {
+    with({ loaded: true, config: __config, cleanKey: __cleanKey, ensureConfig: __ensureConfig }) {
+
+return function /*ensureConfig*/() {
+    if (!loaded) {
+        const envConfig = process.env.PULUMI_CONFIG;
```

Comment:
It 'works' in that *if* you were to have this in your environment, we would read out the values of it.  But, that only happens if the user sets this value.

Note: the reason Config works is actually because we serialize the __config global over.